OV Performance Hints (CPU and GPU logic for selecting the actual configs), while AUTO/MULTI are passing them thru) (#6993)
openvinotoolkit/openvino: inference-engine/samples/benchmark_app/README.md

## How It Works

Upon start-up, the application reads command-line parameters and loads a network and inputs (images/binary files) to the specified device.

  **NOTE**: By default, Inference Engine samples, tools and demos expect input with BGR channels order.
  If you trained your model to work with RGB order, you need to manually rearrange the default channels order in the sample or demo application
  or reconvert your model using the Model Optimizer tool with `--reverse_input_channels` argument specified.
  For more information about the argument, refer to **When to Reverse Input Channels** section of
  [Converting a Model Using General Conversion Parameters](../../../docs/MO_DG/prepare_model/convert_model/Converting_Model_General.md).

> Device specific execution parameters (number of streams, threads, etc) can be either specified explicitly thru the command-line,

Reviewer: ```suggestion
Device-specific execution parameters (number of streams, threads, and so on) can be either explicitly specified through the command line
```

> or left default (and the sample's logic will select the values for the optimal throughput).

Reviewer: ```suggestion
or left default. In the last case, the sample logic will select the values for the optimal throughput.
```

> While playing individual parameters allows to find the performance sweet-spot, usually the parameters are not very performance portable,

Reviewer: ```suggestion
While experimenting with individual parameters allows to find the performance sweet spot, usually, the parameters are not very performance-portable,
```

Reviewer: I'd suggest replacing the word playing with either experimenting or using

so the values from one machine or device are not necessarily optimal for another.

> From that perspective, the most 'portable' way is playing only the 'performance hints', please see section on the command-line params below.

Reviewer: ```suggestion
From this perspective, the most portable way is experimenting only with the performance hints. To learn more, refer to the section on the command-line parameters below.
```

A number of execution steps is defined by one of the following parameters:
* Number of iterations specified with the `-niter` command-line argument
* Time duration specified with the `-t` command-line argument
* Both of them (execution will continue until both conditions are met)
* Predefined duration if `-niter` and `-t` are not specified. Predefined duration value depends on a device.

During the execution, the application calculates latency (if applicable) and overall throughput:
* By default, the median latency value is reported

> * Throughput is calculated as overall_inference_time/number_of_processed_requests. Notice that the throughput value also depends on batch size.

Reviewer: ```suggestion
* Throughput is calculated as overall_inference_time/number_of_processed_requests. Note that the throughput value also depends on batch size.
```

./benchmark_app -m <model> -i <input> -d CPU
```

> But it is still may be sub-optimal for some cases, especially for very small networks. More details can read in [Introduction to Performance Topics](../../../docs/IE_DG/Intro_to_Performance.md).

Reviewer: ```suggestion
But it is still may be sub-optimal for some cases, especially for very small networks. For more details, refer to [Introduction to Performance Topics](../../../docs/IE_DG/Intro_to_Performance.md).
```

    -l "<absolute_path>"        Required for CPU custom layers. Absolute path to a shared library with the kernels implementations.
          Or
    -c "<absolute_path>"        Required for GPU custom kernels. Absolute path to an .xml file with the kernels description.
    -hint "<throughput(or just 'tput')/latency">
                                Optional. Performance hint (optimize for latency or throughput).

>                                 The hint let the OpenVINO device to select the right (network-specific) settings,

Reviewer: ```suggestion
                                The hint allows the OpenVINO device to select the right network-specific settings,
```

>                                 as opposite to just accepting specific values (from the sample's the command-line).

Reviewer: ```suggestion
                                as opposite to just accepting specific values from the sample command line.
```

>                                 So you can specify only the hint without setting  explicit 'nstreams' or other device-specific options.

Reviewer: ```suggestion
                                So you can specify only the hint without setting explicit 'nstreams' or other device-specific options.
```
